libs/sdk-core/src/swap.rs

```diff
@@ -434,6 +434,31 @@ impl BTCReceiveSwap {
         self.swapper_api.complete_swap(payreq.clone()).await
     }
 
+    pub(crate) async fn prepare_refund_swap(
+        &self,
+        req: PrepareRefundRequest,
+    ) -> Result<PrepareRefundResponse> {
+        let swap_info = self
+            .persister
+            .get_swap_info_by_address(req.swap_address.clone())?
+            .ok_or_else(|| anyhow!(format!("swap address {} was not found", req.swap_address)))?;
+
+        let transactions = self
+            .chain_service
+            .address_transactions(req.swap_address.clone())
+            .await?;
+        let utxos = get_utxos(req.swap_address, transactions)?;
```

Comment:
Perhaps we can extract out this code to a function that can be reused in `refund_swap` function?

Reply:
Makes sense. Addressed in [a2308ba](https://github.com/breez/breez-sdk/pull/548/commits/a2308ba88b4b60bc483a1187c071ce5224624c6b)